tests/add.bats

```diff
@@ -392,3 +392,74 @@ EOF
   run_buildah copy --chmod=0755 "$cid" $TEST_SCRATCH_DIR/context/check-dates.sh /
   run_buildah run "$cid" sh -x /check-dates.sh
 }
+
+@test "add-link-flag" {
+  createrandom ${TEST_SCRATCH_DIR}/randomfile
+  createrandom ${TEST_SCRATCH_DIR}/other-randomfile
+
+  run_buildah from $WITH_POLICY_JSON scratch
+  cid=$output
+  run_buildah mount $cid
+  root=$output
+  
+  run_buildah config --workingdir=/ $cid
+  run_buildah add --link $cid ${TEST_SCRATCH_DIR}/randomfile
+  
+  mkdir $root/subdir
+  run_buildah add --link $cid ${TEST_SCRATCH_DIR}/randomfile ${TEST_SCRATCH_DIR}/other-randomfile /subdir
+  
+  run_buildah unmount $cid
+  run_buildah commit $WITH_POLICY_JSON $cid containers-storage:add-link-image
+  run_buildah rm $cid
+
+  run_buildah inspect --type=image add-link-image
+  layers=$(echo "$output" | jq -r '.OCIv1.rootfs.diff_ids | length')
+  if [ "$layers" -lt 2 ]; then
+    echo "Expected at least 2 layers from 2 --link operations, but found $layers"
+    echo "Layers found:"
+    echo "$output" | jq -r '.OCIv1.rootfs.diff_ids[]'
+    exit 1
+  fi
+
+  run_buildah from $WITH_POLICY_JSON add-link-image
+  newcid=$output
+  run_buildah mount $newcid
+  newroot=$output
+  
+  test -s $newroot/randomfile
+  cmp ${TEST_SCRATCH_DIR}/randomfile $newroot/randomfile
+  test -s $newroot/subdir/randomfile
+  cmp ${TEST_SCRATCH_DIR}/randomfile $newroot/subdir/randomfile
+  test -s $newroot/subdir/other-randomfile
+  cmp ${TEST_SCRATCH_DIR}/other-randomfile $newroot/subdir/other-randomfile
+  
+  run_buildah rm $newcid
```

Comment:
Unlike in podman's tests, we wipe out the test's storage area when the test completes, so explicit removal of images and containers is not necessary.